sklearn/utils/testing.py

```diff
@@ -773,3 +773,121 @@ def __init__(self, check, arg_text):
 
     def __call__(self, *args, **kwargs):
         return self.check(*args, **kwargs)
+
+# Utils to test docstrings
+
+def _get_args(function, varargs=False):
+    """Helper to get function arguments"""
+    # NOTE this works only in python3.5
+    if sys.version_info < (3, 5):
```

Comment:
```py

In [1]: from sklearn.pipeline import Pipeline; import inspect

In [2]: import sys; sys.version_info
Out[2]: sys.version_info(major=2, minor=7, micro=12, releaselevel='final', serial=0)

In [3]: hasattr(inspect, 'signature')
Out[3]: False

In [4]: inspect.getargspec(Pipeline.score)
Out[4]: ArgSpec(args=[], varargs='args', keywords='kwargs', defaults=None)

In [5]: 

```

It should work. But it returns an empty list for args of `Pipeline.score`. Because it is wrapped by our `if_delegate_has_method`.

Python 3.5 gives us nice way to unwrap it via the `__wrapped__`, but to do so in python 2.7, it's quite hacky and would involve -

```py
from types import FunctionType
from sklearn.pipeline import Pipeline
from sklearn.utils.metaestimators import _IffHasAttrDescriptor
import inspect

def undecorate(func):
    funcs = []
    
    # python 3.5
    if hasattr(func, '__wrapped__'):
        while hasattr(func, '__wrapped__'):
            func = func.__wrapped__
        return func

    # For py 2.7
    elif hasattr(func, '__closure__'):
        closures = list(c.cell_contents for c in func.__closure__
                        if c.cell_contents is not None)

        # Let's try returning the first closure that is a function object
        funcs = list(c for c in closures if isinstance(c, FunctionType))
        
        # Probably it is decorated with our _IffHasAttrDescriptor
        if len(funcs) == 0:
            funcs = list(c.fn for c in closures if isinstance(c, _IffHasAttrDescriptor))

        return funcs[0]

    elif len(funcs) == 0: 
        raise NotImplementedError("Sorry you are out of luck. I don't handle this func")
    

inspect.getargspec(undecorate(Pipeline.score))
```